src/main.rs

```diff
@@ -373,42 +391,52 @@ fn test_merge_avgs() {
+                    }
+                    t.reset().expect("term::stdout failed");
                 } else {
-                    t.fg(term::color::RED).expect("term::color::RED failed");
                     match options.format {
-                        Some(ref f) => writeln!(
-                            t,
-                            "{}",
-                            f.replace("%n", pkg.as_str())
-                                .replace("%c", avg.issues.iter().join(",").as_str(),)
-                        )
-                        .expect("term::writeln failed"),
+                        Some(ref f) => {
+                            t.fg(term::color::RED).expect("term::color::RED failed");
+                            writeln!(
+                                t,
+                                "{}",
+                                f.replace("%n", pkg.as_str()).replace(
+                                    "%c",
+                                    avg.issues.iter().map(|issue| &issue.0).join(",").as_str(),
+                                )
+                            )
+                            .expect("term::writeln failed");
+                            t.reset().expect("term::stdout failed");
+                        }
                         None => {
-                            let msg = format!(
-                                "Package {} is affected by {}. {}!",
-                                pkg,
-                                avg.issues.join(", "),
-                                avg.severity
-                            );
-
+                            print_avg_colored(&mut t, pkg, avg);
+                            // Colored update
                             if avg.status == enums::Status::Testing {
-                                writeln!(t, "{} Update to {} from the testing repos!", msg, v)
-                                    .expect("term::writeln failed")
+                                // Print: Update to {} for testing repos!"
+                                print!(" Update to");
+                                t.attr(term::Attr::Bold).expect("term::attr failed");
+                                t.fg(term::color::GREEN).expect("term::fg failed");
+                                write!(t, " {}", v).expect("term::writeln failed");
+                                t.reset().expect("term::stdout failed");
+                                println!(" from the testing repos!");
                             } else if avg.status == enums::Status::Fixed {
-                                writeln!(t, "{} Update to {}!", msg, v)
-                                    .expect("term::writeln failed")
-                            } else {
-                                writeln!(t, "{}", msg).expect("term::writeln failed")
+                                // Print: Update to {}!
+                                print!(" Update to");
```

Comment:
I would like this to be a mix of `write!`, `t.reset` and `writeln!` without using `print`.

Reply:
OK, fatto.